### PR TITLE
*: go vet, go lint fixes

### PR DESCRIPTION
**Pull request by gyuho**
Go tip vet complains:

> gopath/src/github.com/coreos/etcd/etcdmain/serve.go:84: assignment copies lock value to dtls: crypto/tls.Config contains sync.Once contains sync.Mutex

https://travis-ci.org/coreos/etcd/jobs/138364239#L810

### CODE REVIEW

**Comment by gyuho** [comment]
still complaining about other issues. Will fix.

**Comment by gyuho** [comment]
/cc @heyitsanthony @xiang90 Please review.

Now it doesn't complain https://travis-ci.org/coreos/etcd/jobs/138427905.

Thanks.

**Comment by heyitsanthony** [comment]
lgtm

**Comment by gyuho** [comment]
Thanks. Merging in greens.

File: etcdmain/etcd.go [hunk 1]
```diff
@@ -267,13 +267,22 @@ func startEtcd(cfg *config) (<-chan struct{}, error) {
...
+		ctlscfgInsecure *tls.Config
+	)
 	if !cfg.clientTLSInfo.Empty() {
 		plog.Infof("clientTLS: %s", cfg.clientTLSInfo)
 		ctlscfg, err = cfg.clientTLSInfo.ServerConfig()
 		if err != nil {
 			return nil, err
 		}
+
+		ctlscfgInsecure, err = cfg.clientTLSInfo.ServerConfig()
```
Comment (xiang90) [inline etcdmain/etcd.go]: there is no other easy way to clone a tlsconf? this makes the func args confusing.

Comment (gyuho) [inline etcdmain/etcd.go]: @xiang90 Yeah it's a big struct https://golang.org/pkg/crypto/tls/#Config. Seems easier just to create another one?

File: pkg/transport/listener.go [hunk 2]
```diff
@@ -252,3 +252,32 @@ func (info TLSInfo) ClientConfig() (*tls.Config, error) {
...
 	return cfg, nil
 }
+
+// CopyTLSConfig copies *tls.Config. This is only
+// work-around for go-vet tests, which complains
+//
+//   assignment copies lock value to p: crypto/tls.Config contains sync.Once contains sync.Mutex
+//
+// Keep up-to-date with 'go/src/crypto/tls/common.go'
+func CopyTLSConfig(cfg *tls.Config) *tls.Config {
```
Comment (xiang90) [inline pkg/transport/listener.go]: ShallowCopyTLSConfig?

**Comment by gyuho** [comment]
@xiang90 Just fixed.

And added more fixes for go lint issues. https://goreportcard.com/report/github.com/coreos/etcd#golint

/cc @heyitsanthony

**Comment by heyitsanthony** [comment]
lgtm

**Comment by xiang90** [comment]
lgtm

**Comment by gyuho** [comment]
thanks.

**Pull request merged**